lib/active_model/serializer.rb

```diff
@@ -84,8 +84,10 @@ def self.serializer_for(resource, options = {})
         resource.serializer_class
       elsif resource.respond_to?(:to_ary)
         config.array_serializer
+      elsif options.key?(:serializer)
+        options[:serializer]
       else
-        options.fetch(:serializer) { get_serializer_for(resource.class) }
+        get_serializer_for(resource.class, options[:parent_serializer], options[:controller])
```

Comment:
Yeah, I think the whole `serializer_for` calling `get_serializer_for` thing is pretty bad design already. I'm really open to suggestions to make the whole serializer lookup cleaner.